sdk/core/azure-core/inc/http/policy.hpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "context.hpp"
+#include "http.hpp"
+
+namespace Azure
+{
+namespace Core
+{
+namespace Http
+{
+
+class HttpPolicy
+{
+public:
+  // If we get a response that goes up the stack
+  // Any errors in the pipeline throws an exception
+  // At the top of the pipeline we might want to turn certain responses into exceptions
+  virtual Response Process(Context& context, Request& request) const = 0;
```

Comment:
lvalue-reference-to-non-`const` rejects temporaries. Is that OK for context or should this take a copy?

Reply:
We want this to refer to the Context passed in. The caller (or another thread) could cancel it and this policy MUST be able to see that cancellation.